rust/agama-server/src/web/http.rs

```diff
@@ -90,3 +92,61 @@ pub async fn logout(_claims: TokenClaims) -> Result<impl IntoResponse, AuthError
 pub async fn session(_claims: TokenClaims) -> Result<(), AuthError> {
     Ok(())
 }
+
+// builds a response tuple for translation redirection
+fn redirect_to_file(file: &str) -> (StatusCode, HeaderMap, Body) {
+    tracing::info!("Redirecting to translation file {}", file);
+
+    let mut response_headers = HeaderMap::new();
+    // translation found, redirect to the real file
+    response_headers.insert(
+        header::LOCATION,
+        // if the file exists then the name is a valid value and unwrapping is safe
+        HeaderValue::from_str(file).unwrap(),
+    );
+
+    (
+        StatusCode::TEMPORARY_REDIRECT,
+        response_headers,
+        Body::empty(),
+    )
+}
+
+// handle the /po.js request
+// the requested language (locale) is sent in the "agamaLang" HTTP cookie
+// this reimplements the Cockpit translation support
+pub async fn po(State(state): State<ServiceState>, jar: CookieJar) -> impl IntoResponse {
+    if let Some(cookie) = jar.get("agamaLang") {
+        tracing::info!("Language cookie: {}", cookie.value());
+        // try parsing the cookie
+        if let Some((lang, region)) = cookie.value().split_once('-') {
+            // first try language + country
+            let target_file = format!("po.{}_{}.js", lang, region.to_uppercase());
+            if state.public_dir.join(&target_file).exists() {
+                return redirect_to_file(&target_file);
+            } else {
+                // then try the language only
+                let target_file = format!("po.{}.js", lang);
+                if state.public_dir.join(&target_file).exists() {
+                    return redirect_to_file(&target_file);
+                };
+            }
+        } else {
+            // use the cookie as is
+            let target_file = format!("po.{}.js", cookie.value());
+            if state.public_dir.join(&target_file).exists() {
+                return redirect_to_file(&target_file);
```

Comment:
I see three times almost identical code. Maybe you can use something like

```
fn redirect_to_file(state, file) -> Option<impl IntoResponse>

// and here in code use something like
if Some(response) = redirect_to_file(&state, &file) {
  return response
} 
```

this way logic with testing file existence in public dir is implemented just once.

Reply:
maybe name can be `serve_file_if_exist`

Reply:
Tried that, it did not look any better, the added Some/None handling made the code less readable IMHO...